gpu/impl/Distance.cu

```diff
@@ -19,6 +19,7 @@
 #include "../utils/BlockSelectKernel.cuh"
 
 #include <memory>
+#include <algorithm>
```

Comment:
What is the rationale behind adding this header?

Reply:
To my understanding, this file is using at least `std::min`, which is defined by this header. Can't quite justify why GCC can still compile without this line though.